ibm/resource_ibm_private_dns_record.go

```diff
@@ -0,0 +1,193 @@
+package ibm
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	//"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	iserrors "github.ibm.com/Bluemix/riaas-go-client/errors"
+)
+
+var allowedPrivateDomainRecordTypes = []string{
+	"A", "AAAA", "CNAME", "MX", "PTR", "SRV", "TXT",
+}
+
+const (
+	pdnsRecordType  = "type"
+	pdnsRecordTTL   = "ttl"
+	pdnsIPv4Address = "ipv4_address"
+	pdnsRecordName  = "name"
+)
+
+func resourceIBMPrivateDNSRecords() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMPrivateDnsRecordCreate,
+		Read:     resourceIBMPrivateDnsRecordRead,
+		Update:   resourceIBMPrivateDnsRecordUpdate,
+		Delete:   resourceIBMPrivateDnsRecordDelete,
+		Exists:   resourceIBMPrivateDnsRecordExists,
+		Importer: &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		CustomizeDiff: customdiff.Sequence(
+			func(diff *schema.ResourceDiff, v interface{}) error {
+				return resourceTagsCustomizeDiff(diff)
+			},
+		),
+
+		Schema: map[string]*schema.Schema{
+			pdnsInstanceID: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+
+			pdnsZoneID: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+
+			pdnsRecordTTL: {
+				Type:     schema.TypeInt,
+				Required: true,
+				DefaultFunc: func() (interface{}, error) {
+					return 900, nil
+				},
+			},
+
+			pdnsRecordName: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			pdnsIPv4Address: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			pdnsRecordType: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: func(val interface{}, field string) (warnings []string, errors []error) {
+					value := val.(string)
+					for _, rtype := range allowedPrivateDomainRecordTypes {
+						if value == rtype {
+							return
+						}
+					}
+
+					errors = append(
+						errors,
+						fmt.Errorf("%s is not one of the valid domain record types: %s",
+							value, strings.Join(allowedPrivateDomainRecordTypes, ", "),
+						),
+					)
+					return
+				},
+			},
+		},
+	}
```

Comment:
For different types of record types we have different rdata
https://cloud.ibm.com/apidocs/dns-svcs#create-a-resource-record 
https://github.com/IBM-Cloud/terraform-provider-ibm/blob/06df6ca39a6e811795cb4d7d4441125d7ac5add2/ibm/resource_ibm_cis_dns_record.go#L50